src/logistics/TerminalNetwork_v2.ts

```diff
@@ -911,7 +912,7 @@ export class TerminalNetworkV2 implements ITerminalNetwork {
 	private handleProvideInstance(colony: Colony, resource: ResourceConstant, provideAmount: number,
 								  partnerSets: Colony[][], opts: ProvideOpts): boolean {
 		// Sometimes we don't necessarily want to push to other rooms - we usually do, but not always
-		if (opts.allowPushToOtherRooms) {
+		if (opts.allowPushToOtherRooms && colony.terminal!.store[resource] > this.lockedAmount(colony, resource)) {
```

Comment:
`lockedAmount` is based on the colony assets, not the amount that is currently in the terminal. In like 90% of cases, anything that looks at whether a colony has some amount of a resource should use `colony.assets`, not direct terminal or storage lookups

Reply:
The bug is that `sendAmount` is restricted to `sendTerm.store[resource] - this.lockedAmount(colony, resource)`, which means that if this check isn't here, that value is negative and it tries to send a negative amount, which seems pointless. So, while I'd normally agree that `colony.assets` should be used mostly, when you're calculating whether you can issue a send on the current tick, the only thing that matters is how much is in the terminal you want to use right now.